pkg/kubelet/util/format/pod.go

```diff
@@ -57,22 +57,13 @@ func PodWithDeletionTimestamp(pod *v1.Pod) string {
 	return Pod(pod) + deletionTimestamp
 }
 
-// Pods returns a string representation a list of pods in a human
-// readable format.
-func Pods(pods []*v1.Pod) string {
-	return aggregatePods(pods, Pod)
-}
-
-// PodsWithDeletionTimestamps is the same as Pods. In addition, it prints the
-// deletion timestamps of the pods if they are not nil.
-func PodsWithDeletionTimestamps(pods []*v1.Pod) string {
-	return aggregatePods(pods, PodWithDeletionTimestamp)
-}
-
-func aggregatePods(pods []*v1.Pod, handler podHandler) string {
-	podStrings := make([]string, 0, len(pods))
-	for _, pod := range pods {
-		podStrings = append(podStrings, handler(pod))
+// Pods returns a list of pods as ObjectRef
+func Pods(pods []*v1.Pod) []klog.ObjectRef {
```

Comment:
There are two questions related to changing this function:
1. Is this a good place to have `Pods` formatter if `KObj` is in `klog`?
2. How will this effect existing users of `format.Pods` that still use `Infof`?

For first I think it's ok for temporary solution, but we should move this function to klog as it affects any list of K8s objects. With current time before code freeze I think we will not have enough time to move it, but we should plan it for future.

2. Could you either check if there are no left format.Pods uses that pass it to `klog.Infof` **or** show examples how output will look like with `Infof`.

If there are some cases with `Infof` and the result is not satisfactory (names of pods&namespaces are not easily readable), then I think we should migrate them to InfoS before merging this PR.

Reply:
> How will this effect existing users of format.Pods that still use Infof?

I think we should migrate all those as part of this PR if there are any remaining, which I think addresses your other concern. I looked at the use of this formatter and it was limited to one or two files in kubelet, I'm surprised it got factored out into util.

> For first I think it's ok for temporary solution, but we should move this function to klog as it affects any list of K8s objects. With current time before code freeze I think we will not have enough time to move it, but we should plan it for future.

Yeah ideally I'd much rather update `klog.KObj`/have something in klog that, given a slice of objects, returns a slice of `klog.ObjectRef` but @DirectXMan12 told me this isn't currently supported, so I think we should just do this for now.

@QiWang19 can you check to make sure that no instances of `format.Pods` are called with non-structured logging?

Reply:
> @QiWang19 can you check to make sure that no instances of `format.Pods` are called with non-structured logging?

pkg/kubelet/kubelet.go and pkg/kubelet/eviction/eviction_manager.go are using `format.Pods`. 
kubelet.go is non-structured, eviction_manager.go is worked in PR https://github.com/kubernetes/kubernetes/pull/99032/

Reply:
@navidshaikh is working on kubelet.go, which is quite large https://github.com/kubernetes/kubernetes/issues/98976#issuecomment-791499864 Let's coordinate, @QiWang19 I suggest you migrate *only* the affected lines in this PR and Navid can take care of the rest.

Reply:
thanks! https://github.com/kubernetes/kubernetes/pull/99799 migrates the rest of `kubelet.go` except `format.Pods` lines. (there are some instances of `format.Pods` *with* structured logging in `pkg/kubelet/eviction/eviction_manager.go`)